### PR TITLE
refactor: use isAutofixable for Autofix [ZEN-707]

### PR DESCRIPTION
**Pull request by michelkaporin**
### Description

Use SARIF property to determine if a certain issue is autofixable instead of relying on filters endpoint.
Do not merge until https://github.com/snyk/sast-analysis-api/pull/455 is through.

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] README.md updated, if user-facing
- [ ] License file updated, if new 3rd-party dependency is introduced

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

**Review by patricia-v — APPROVED** [review]
LGTM! 🚀

File: infrastructure/code/code_test.go [hunk 1]
```diff
@@ -654,14 +654,15 @@ func TestUploadAnalyzeWithAutofix(t *testing.T) {
 	)
 
 	t.Run(
-		"should not run autofix after analysis when is enabled but have disallowed extension",
+		"should not provide autofix code action when autofix enabled but issue not fixable",
```
Comment (bastiandoetsch) [inline infrastructure/code/code_test.go:657]: do you already have a positive test that it's added when fixable?

**Review by bastiandoetsch — APPROVED** [review]

**Review by pkey — APPROVED** [review]
hmm, questionable ownership of @snyk/nebula. Don't feel I can review this adequately but LTGM from what I see. I will raise this on Slack with my team to discuss it.

File: infrastructure/code/code_test.go (hunk 1 above)
Comment (michelkaporin) [reply infrastructure/code/code_test.go:657]: Yes, it's called `should run autofix after analysis when is enabled` in the same test suite 👌

**Comment by michelkaporin** [comment]
Will be merging this tomorrow to give time for complete cache invalidation after 24h on `sast-analysis-api` since https://github.com/snyk/sast-analysis-api/pull/455.

**Pull request merged**